producer/src/did_document.rs

```diff
@@ -3,22 +3,30 @@ use identity_iota::iota::IotaDID;
 use serde::{Deserialize, Serialize};
 use shared::error::ProducerError;
 use shared::JwkStorageWrapper;
+use std::string::ToString;
+use strum::Display;
 
 use crate::SecretManager;
 
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Display)]
 pub enum DidMethod {
     #[serde(rename = "did:jwk")]
+    #[strum(serialize = "did:jwk")]
     Jwk,
     #[serde(rename = "did:key")]
+    #[strum(serialize = "did:key")]
     Key,
     #[serde(rename = "did:web")]
+    #[strum(serialize = "did:web")]
     Web,
     #[serde(rename = "did:iota:rms")]
+    #[strum(serialize = "did:iota:rms")]
     ShimmerTestnet,
     #[serde(rename = "did:iota:smr")]
+    #[strum(serialize = "did:iota:smr")]
     Shimmer,
     #[serde(rename = "did:iota")]
+    #[strum(serialize = "did:iota")]
     IotaMainnet,
 }
```

Comment:
`[resolved]`

Your solution obviously works but leaving this here for your consideration :) 

Instead of adding a new dependency you could consider solving this as follows:
```rust

#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
pub enum DidMethod {
    #[serde(rename = "did:jwk")]
    Jwk,
    #[serde(rename = "did:key")]
    Key,
    #[serde(rename = "did:web")]
    Web,
    #[serde(rename = "did:iota:rms")]
    ShimmerTestnet,
    #[serde(rename = "did:iota:smr")]
    Shimmer,
    #[serde(rename = "did:iota")]
    IotaMainnet,
}

// Through this trait you obtain the ability to call `to_string` on your type. Under the hood it simply uses the `Serialize` trait through `serde_json`.
impl std::fmt::Display for DidMethod {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", serde_json::json!(self).as_str().ok_or(std::fmt::Error)?)
    }
}

// A similar strategy can be used to construct your type from a string. Under the hood it simply uses the `Deserialize` trait through `serde_json`.
impl std::str::FromStr for DidMethod {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        serde_json::from_str(&format!("{s:?}")).map_err(|e| e.to_string())
    }
}

#[test]
fn test() {
    let test = DidMethod::Jwk;

    // Using serialization
    // prints: `did:jwk`
    println!("{}", serde_json::json!(test).as_str().unwrap());

    let string = test.to_string();

    // Using `to_string`
    // prints: `did:jwk`
    println!("{}", string);

    // `from_str` uses the `Deserialize` trait under the hood
    let method: DidMethod = DidMethod::from_str("did:jwk").unwrap();

    // Jwk
    println!("{:?}", method);
}
```

Added benefit of this strategy is that you don't need to add
```rust
#[serde(rename = "did:iota:smr")]
#[strum(serialize = "did:iota:smr")]
```
for each variant.